nemo/collections/asr/parts/submodules/jasper.py

```diff
@@ -958,3 +958,80 @@ def forward(self, input_: Tuple[List[Tensor], Optional[Tensor]]):
             return xs + [out], lens
 
         return [out], lens
+
+
+class ParallelBlock(nn.Module):
+    """
+    Computational module that computes several `blocks` independently from each other and aggregates the outputs.
+    Due to a particular way of structuring inputs with sequence masks in Jasper-like blocks, this module focuses
+    specifically on the Jasper family.
+
+    Args:
+        blocks: List of Jasper blocks that will be computed concurently. It is expected that they accept the same
+            input and return outputs with the same number of channels.
+        aggregation_mode: an optional string, indicating how the outputs will be aggregated. Supported values are
+            ['sum', 'dropout']. "sum" value forces outputs to be summed together. "dropout" value enables tower
+            dropout training with different blocks being dropped out during training.
+        block_dropout_prob: a probability of dropping any individual block during training with "dropout" aggregation
+            mode. Acts as a regularization technique.
+    """
+
+    def __init__(self, blocks, aggregation_mode: str = "sum", block_dropout_prob: int = 0.0):
+        super().__init__()
+        self.blocks = nn.ModuleList(blocks)
+        self.aggregation_mode = aggregation_mode
+        if aggregation_mode == "dropout":
+            self.weights = nn.Parameter(torch.ones(1, len(blocks), 1, 1), requires_grad=False)
+            self.dropout = nn.Dropout(block_dropout_prob)
+
+    def get_dropout_mask(self):
+        weights = self.dropout(self.weights)
+        while torch.sum(weights) == 0 and self.dropout.p < 1.0:
```

Comment:
Why is this while loop here? It is fine if dropout(weights) doesnt dropout anything for a few steps

Reply:
This loop prevents the inverse situation: dropout discarding all towers. Since the number of towers is low, this can happen quite often. 
Since the residual is there anyway, the model will learn even with all towers discarded, emulating stochastic depth behaviour, but we want at least one tower to learn something at each iteration.

Reply:
Ah that makes sense. Alright let's leave it as is.